libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/util/CacheResultUtils.kt

```diff
@@ -126,4 +244,18 @@ internal object CacheResultUtils {
             }
         }
     }
+
+    private fun <T> listElementsAreEqual(
+        first: List<T>,
+        second: List<T>,
+        equalityFun: (T, T) -> Boolean
+    ): Boolean {
+        if (first.size != second.size) {
+            return false
+        }
+
+        return first.zip(second).all { (x, y) ->
```

Comment:
Zip will create an intermediate collection. Do we accept that? I mean this PR is about optimizations. I think not using an intermediate collection might optimize it further.

Reply:
This isn't close to the bottleneck.  The real optimization is reducing the amount of the routes that need to be compared to determine equality for this specific use case.